src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -128,6 +128,9 @@ public void Add(X509Certificate2 certificate)
         {
             if (certificate == null)
                 throw new ArgumentNullException(nameof(certificate));
+            
+            if (certificate.Handle == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(certificate));
```

Comment:
Alright, I changed it again to throw a CryptographicException and I changed the order in which we throw exceptions as well.